File: web-nodejs/extension-common/src/model/FlexMediaDetail.ts
import {FlexMediaUser} from "./FlexMediaUser";
import {FlexMediaPlaylist} from "./FlexMediaPlaylist";
import {FlexMediaDetailSeries} from "./FlexMediaDetailSeries";
import {FlexMediaTag} from "./FlexMediaTag";
import {FlexMediaDetailTab} from "./FlexMediaDetailTab";

/**
 * 条目详细信息
 *
 * @author why
 * @since 5/8/24
 */
export interface FlexMediaDetail {
    /**
     * 条目 ID
     */
    id: string

    /**
     * 条目标题
     */
    title: string

    /**
     * 描述
     */
    description: string

    /**
     * 封面
     */
    cover: string

    /**
     * 分享链接
     */
    url: string

    /**
     * 类型描述
     */
    type: string | undefined

    /**
     * 播放次数
     */
    playCount: string | undefined

    /**
     * 时间
     */
    createAt: string | undefined

    /**
     * 时长
     */
    duration: number | undefined

    /**
     * 大小
     */
    size: string | undefined

    /**
     * 发布者
     */
    publisher: FlexMediaUser | undefined

    /**
     * 包含的播放列表
     */
    playlist: FlexMediaPlaylist[] | undefined

    /**
     * 该条目的系列数据
     */
    series: FlexMediaDetailSeries[] | undefined

    /**
     * 条目的标签
     */
    tags: FlexMediaTag[] | undefined

    /**
     * 条目下方的 TAB UI配置
     */
    relativeTabs: FlexMediaDetailTab[] | undefined

    /**
     * 条目的扩展信息
     */
    extras: Map<string, string> | undefined
}

File: lib-extension/src/main/java/com/xiaoyv/flexiflix/extension/model/FlexMediaDetail.kt
package com.xiaoyv.flexiflix.extension.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import com.xiaoyv.flexiflix.extension.utils.UNKNOWN_STRING
import kotlinx.parcelize.Parcelize

/**
 * [FlexMediaDetail]
 *
 * @author why
 * @since 5/8/24
 */
@Parcelize
data class FlexMediaDetail(
    @SerializedName("id") var id: String,
    @SerializedName("title") var title: String,
    @SerializedName("description") var description: String,
    @SerializedName("cover") var cover: String,
    @SerializedName("url") var url: String,
    @SerializedName("type") var type: String? = UNKNOWN_STRING,
    @SerializedName("playCount") var playCount: String? = UNKNOWN_STRING,
    @SerializedName("createAt") var createAt: String? = UNKNOWN_STRING,
    @SerializedName("duration") var duration: Long = 0,
    @SerializedName("size") var size: String? = UNKNOWN_STRING,
    @SerializedName("publisher") var publisher: FlexMediaUser? = FlexMediaUser(),
    @SerializedName("playlist") var playlist: List<FlexMediaPlaylist>? = emptyList(),
    @SerializedName("series") var series: List<FlexMediaDetailSeries>? = emptyList(),
    @SerializedName("tags") var tags: List<FlexMediaTag>? = emptyList(),
    @SerializedName("relativeTabs") var relativeTabs: List<FlexMediaDetailTab>? = emptyList(),
    @SerializedName("extras") var extras: HashMap<String, String>? = hashMapOf()
) : Parcelable